### PR TITLE
Fix test failures with Pandas 1.2.0

### LINKED ISSUE #158: Support Pandas version 1.2.0
Currently there are test failures when using Pandas 1.2.0. Need to resolve these to fully support Pandas 1.2.0 and above.

### PR DESCRIPTION
**Pull request by BryanCutler**
There are a number of test failures when using Pandas 1.2.0.

* The method `pandas.core.ops._get_op_name()` has been removed in Pandas >=1.2.0. This was being used when building `TensorArray` ops, and is replaced with similar usage in `ExtensionScalarOpsMixin`.
* `TensorArray` should return it's instance when `astype()` called with same type
* `TensorArray._contains_` needs to have proper implementation of `any()`
* `repr()` is broken when using float values of ndims > 2

Fixes #158

### CODE REVIEW

**Comment by BryanCutler** [comment]
There seem to be a number of different issue with Pandas, I'll look into fixing them here too.

**Comment by frreiss** [comment]
@BryanCutler any idea what's causing that test failure on the Python 3.7 build (https://github.com/CODAIT/text-extensions-for-pandas/pull/157/checks?check_run_id=1620791768)?

**Comment by BryanCutler** [comment]
I fixed most of the issues with pandas 1.2.0 and TensorArray, except there is a regression with `repr()` with floats, similar to #151. There has been some back and forth with my fix upstream for that one, so maybe I can fix both. Unfortunately, I don't know of a workaround, so we either have to leave it as a known issue or cap the supported pandas version < 1.2.0.  I'm looking into the other issues now.

File: text_extensions_for_pandas/io/watson/test_tables.py [hunk 1]
```diff
@@ -445,6 +446,8 @@ def test_make_exploded_df(self):
 15        Total tax rate  \
 """)
 
+    @pytest.mark.skipif(LooseVersion(pd.__version__) >= LooseVersion("1.2.0"),
+                        reason="TODO: Rank col gets converted to float")
```
Comment (BryanCutler) [inline text_extensions_for_pandas/io/watson/test_tables.py:450]: Not sure why this is different now, but I think it's safe to skip and I can make an issue to follow up with later

**Comment by BryanCutler** [comment]
@frreiss this failure https://github.com/CODAIT/text-extensions-for-pandas/pull/157/checks?check_run_id=1646927381#step:5:1438 in `bert.align_bert_tokens_to_corpus_tokens()` seems a bit strange. Not sure why this would suddenly change, unless they added some optimization that wasn't complete. I can look into it further, but letting you know in case you have some idea what's going on.

**Comment by BryanCutler** [comment]
Ok, I think I found why the above error is happening and looks to be a bug. Pandas attempts to do a cython aggregation and then fallback to a vanilla agg if there is an error, but checks the error message improperly for this case. I should be able to file a bug report and do a PR for the fix.

**Comment by frreiss** [comment]
Thanks for tracking down the root cause of that problem, @BryanCutler !

**Comment by BryanCutler** [comment]
PR for the agg fix https://github.com/pandas-dev/pandas/pull/38982

**Comment by BryanCutler** [comment]
@frreiss I capped the upper version of Pandas to < 1.2.0 because not being able to display a Series with TensorArray of floats is pretty major and I don't see a workaround. I'll keep working on the upstream fix, but what are your thoughts on merging this?

File: text_extensions_for_pandas/io/watson/test_tables.py (hunk 1 above)
Comment (frreiss) [reply text_extensions_for_pandas/io/watson/test_tables.py:450]: Works for me.

**Comment by frreiss** [comment]
I think merging is the best way forward for now. We may want to post a new release just to get that pandas<1.2.0 constraint into our requirements.txt.

**Pull request merged**